xrpl/ledger-entry-types/oracle.go

```diff
@@ -19,6 +40,53 @@ type PriceData struct {
 	Scale uint8 `json:",omitempty"`
 }
 
+// Validate validates the price data.
+func (priceData *PriceData) Validate() error {
+	if len(priceData.BaseAsset) == 0 {
+		return ErrPriceDataBaseAsset
+	}
+
+	if len(priceData.QuoteAsset) == 0 {
+		return ErrPriceDataQuoteAsset
+	}
+
+	if priceData.Scale > PriceDataScaleMax {
+		return ErrPriceDataScale
+	}
+
+	if (priceData.AssetPrice == 0) != (priceData.Scale == 0) {
+		return ErrPriceDataAssetPriceAndScale
+	}
+
+	return nil
+}
+
+// Flatten flattens the price data.
+func (priceData *PriceData) Flatten() map[string]interface{} {
+	mapKeys := 2
+
+	if priceData.Scale != 0 && priceData.AssetPrice != 0 {
+		mapKeys = 4
+	}
+
+	flattened := make(map[string]interface{}, mapKeys)
+
+	if priceData.BaseAsset != "" {
+		flattened["BaseAsset"] = priceData.BaseAsset
+	}
+	if priceData.QuoteAsset != "" {
+		flattened["QuoteAsset"] = priceData.QuoteAsset
+	}
+	if priceData.AssetPrice != 0 {
+		flattened["AssetPrice"] = priceData.AssetPrice
+	}
+	if priceData.Scale != 0 {
```

Comment:
What happens if Scale = 0 (which is a valid value)?

According to the docs:

```
The scaling factor to apply to an asset price. For example, if Scale is 6 and original price is 0.155, then the scaled price is 155000. Valid scale ranges are 0-10. It's not included if the last update transaction didn't include the BaseAsset/QuoteAsset pair.
```

Reply:
Fixed, I added a comment explaining that `Scale` default value is `0`